lib/mapper/product_mapper.rb

```diff
@@ -140,6 +142,19 @@ def self.get_categories(product)
                 return categories
             end
 
+            def self.get_categories_after_split(product, categories)
+
+                if product['model'].select { |c| c['categories']}
+                    product['model'].each do |variant|
+                        variant['categories'].map do |category|
+                            categories.push(category)
+                        end
+                    end
+                end
+
+                return categories
+            end
```

Comment:
I don't think this is right.

At this point in the process, `product` would have all the categories across _all variants_. As is, this would perpetuate the problem we're trying to solve.

You need to filter this based onwhether you're getting categories for the digital product or the physical product..

Reply:
@Jhalter5Stones Following our chat -- it sounds like this is actually correct.

We should add a comment indicating that the `categories` array coming into this method contains categories shared by both physical/digital products.